# Text Analysis using Python

## Objective
Analyze a given text and calculate the number of words, characters, vowels, and repeated words.

In [2]:
from collections import Counter

# Sample text
text = """
Natural Language Processing is a field of artificial intelligence.
Natural language processing helps computers understand human language.
Text analysis is an important part of natural language processing.
"""

# Convert text to lowercase
clean_text = text.lower()

# Split text into words
words = clean_text.split()

# Count words
word_count = len(words)

# Count characters
character_count = len(text)

# Count vowels
vowels = "aeiou"
vowel_count = sum(1 for char in clean_text if char in vowels)

# Count repeated words
word_frequency = Counter(words)

repeated_words = {
    word: count
    for word, count in word_frequency.items()
    if count > 1
}

# Display results
print("TEXT ANALYSIS RESULTS")
print("---------------------")
print("Number of words:", word_count)
print("Number of characters:", character_count)
print("Number of vowels:", vowel_count)

print("\nRepeated words:")
for word, count in repeated_words.items():
    print(word, ":", count)

TEXT ANALYSIS RESULTS
---------------------
Number of words: 27
Number of characters: 206
Number of vowels: 69

Repeated words:
natural : 3
language : 3
processing : 2
is : 2
of : 2


## Analysis and Results

The text was analyzed using basic Python string operations and the Counter class.

### Results

- *Number of words:* 27
- *Number of characters:* 206
- *Number of vowels:* 69
- *Repeated words:* Natural, language, processing, is, and of

The repeated-word analysis shows which words occur more than once in the given text.

## Conclusion

This project demonstrates basic text-processing techniques using Python.
The program successfully calculates word count, character count, vowel count,
and identifies repeated words. These techniques are useful as basic building
blocks for Natural Language Processing (NLP).

## Interview Questions

### 1. What is text preprocessing?
Text preprocessing is the process of cleaning and preparing raw text before analyzing it or using it in a Natural Language Processing system. Common steps include converting text to lowercase, removing unnecessary characters, tokenization, and removing stop words.

### 2. How can repeated words be identified?
Repeated words can be identified by counting the frequency of each word. Python's Counter from the collections module can be used to count how many times each word occurs.

### 3. What is NLP?
NLP (Natural Language Processing) is a field of Artificial Intelligence that enables computers to understand, process, and analyze human language.